src/compose.ts

```diff
@@ -16,11 +16,67 @@ import {
 const initFnMap = new WeakMap<Function, ComposeInitializationFunction<any, any>[]>();
 
 /**
+ * The default factory label if no label can be derived during the factory creation process
+ */
+const COMPOSE_LABEL = 'Compose';
+
+/**
+ * Reference to defineProperty to support minification
+ */
+const defineProperty = Object.defineProperty;
+
+/**
  * A weakmap that will store static properties for compose factories
  */
 const staticPropertyMap = new WeakMap<Function, {}>();
 
 /**
+ * Internal function which can label a function with a name
+ */
+function labelFunction(fn: Function, value: string): void {
+	const nameDescriptor = Object.getOwnPropertyDescriptor(fn, 'name');
+	if (typeof nameDescriptor === 'undefined' || nameDescriptor.configurable) {
+		defineProperty(fn, 'name', {
+			value,
+			writable: true,
+			configurable: true
+		});
+	}
+}
+
+/**
+ * Internal function which can label a factory with a name and also provides a
+ * function which
+ *
+ * @param fn The name of the factory to label
+ * @param value The name to supply for the label
+ */
+function labelFactory(factory: Function, value: string): void {
+	if (typeof factory === 'function' && factory.prototype) {
+		labelFunction(factory, value);
```

Comment:
Perhaps `setLabel` would be a better function name. Spent like 30 seconds thinking this method would have a return value that should have been used.